### ISSUE TITLE
unprivileged users can edit permissions as root

### ISSUE THREAD

**Issue by corpeder** [webmin/webmin#1492]
So i have the latest webmin running on the latest debian 10.
fresh install of both
after creating a new user without sudo and then creating a webmin user matching the non-sudo account
giving that user file manager access gives them full control of filesystem and does not follow the settings applied in webmin user > permissions for all modules

DOES NOT FOLLOW ROOT DIRECTORY FOR FILE PICKER OR BROWSING FILES AS A SPECIFIC LINUX USER

**Comment by swelljoe** [comment]
Webmin users are always assumed to be root. If you want them to not have those privileges, you need to make use of the access control features of the module in question to limit them. Some modules don't have fine-grained controls and always operate as root.

If you want a user to not be assumed to be root, give them access to Usermin. It has File Manager, too. That's what Usermin is for. Webmin is an administrative tool that is always assumed to be used for administrative tasks.

**Comment by swelljoe** [comment]
To be clear: This is not a bug, it is operating as intended and as documented.

**Comment by swelljoe** [comment]
Oh, wait...maybe I misunderstand what you're saying here. And, maybe it isn't quite as clear that Webmin File Manager is doing the right/documented thing.

**Comment by swelljoe** [comment]
To be clear again: It is still acting as designed/intended, but I don't know that the design/intention is *right*. I can see how you could read those options and think they would apply to File Manager. They don't, and there is no code to make them do so, and they never have...but, I can see how you could read it that way, and probably other people would, too.

**Comment by chris001** [comment]
One way to address this would be to block users being added to file manager access, if users don't have `sudo` ability, with an error message explaining why.  This would make it very clear to any webmin (admin) user that only users with `root` `sudo` powers are allowed using file manager access, because file manager access empowers those users with `root` `sudo` powers, in a sense, they go hand in hand.. Can't have one without the other, otherwise unintended consequences will happen.

**Comment by swelljoe** [comment]
That goes against what Webmin is for. It is an administrative tool. Every user you create in Webmin is for some elevated privilege purpose (including e.g. Virtualmin users, who need to be able to modify service configs like Apache virtual hosts or BIND zones). If we prevented creating any kind of non-sudo user, it would make Webmin useless because it would apply to nearly every function.

The problem here is a usability problem, not a functionality problem, and the solution is not to remove functionality. File Manager *should* be able to be used to allow a non-root user to modify some root files (otherwise they should use Usermin). The problem is that the UI gives the impression that these limits would apply to File Manager (and others...there are several ways I think this page could be interpreted to limit things that it will not limit).

So, either the UI needs to not make the claim that it limits all modules, or it needs to do what it seems like it is offering to do. It is labeled *Permissions for all modules*, which is...misleading, at best. It is configuring a few API functions that are shared across all (well, most) modules, like the file picker...but the File Manager isn't limited by these permissions.

Error messages and preventing a very common use case is not the solution. Fixing the UI is the solution.

**Comment by iliarostovtsev** [comment]
I would like to add that there is a new functionality in Webmin Users module to create a safe Webmin user. Later, using this user's ACLs for the File Manager you could control what paths are accessible.

Besides, latest Webmin available (1.974) also allows you to limit default (privileged) Webmin user allowed paths.

**Comment by iliarostovtsev** [comment]
> after creating a new user without sudo and then creating a webmin user matching the non-sudo account

Yeah, that will work as you expect if you create mentioned earlier safe Webmin user.
